### PR TITLE
start adding git witchcraft

### PR DESCRIPTION
**Pull request by kleebaum**

### CODE REVIEW

File: plugin-src/gitVisualization/gitWitchcraft.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+async function zoomToChanges(absoluteFilePaths: string[]): Promise<void> {
+    const rootFolder: RootFolderBox = getRootFolder()
+    const changedFileBoxesRendered: Box[] = absoluteFilePaths.map(path =>
+        rootFolder.getRenderedBoxesInPath(path).at(-1)).filter(box => box) as Box[]
+    const map: Map = getMapOrError()
+    map.zoomToFitBoxes(changedFileBoxesRendered)
+}
+
+export async function visualizeCommitByRef(ref: string): Promise<void> {
+    visualizeChanges(`${ref}^`, ref, true)
```
Comment (MichiSpebach) [inline plugin-src/gitVisualization/gitWitchcraft.ts:50]: this returns directly and does not await `visualizeChanges`

File: plugin-src/gitVisualization/gitWitchcraft.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+export async function visualizeLastCommit(): Promise<void> {
+    visualizeCommitByRef('HEAD')
```
Comment (MichiSpebach) [inline plugin-src/gitVisualization/gitWitchcraft.ts:54]: this returns directly and does not await `visualizeCommitByRef`

File: plugin-src/gitVisualization.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+})
+applicationMenu.addMenuItemTo('gitVisualization.js', commitByRefMenuItem)
+applicationMenu.addMenuItemTo('gitVisualization.js', lastCommitMenuItem)
+
+async function promptForGitRef(): Promise<void> {
+	const ref: string | undefined = await TextInputPopup.buildAndRenderAndAwaitResolve('Git Ref', 'HEAD')
+	if (ref) {
+		await visualizeCommitByRef(ref)
+	}
+}
```
Comment (MichiSpebach) [inline plugin-src/gitVisualization.ts:21]: maybe we don't need the application menu entries anymore and the sidebar is enough 🤔

**Review by MichiSpebach — APPROVED** [review]

File: plugin-src/gitVisualization/GitVisualizationToolbarView.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+                },
+                {
+                    type: 'td',
+                    innerHTML: '&#129668; Uncommitted changes'
+                }
+            ]
+        }
+    }
+
+    private commitsToToggles(commits: Commit[]): RenderElement[] {
```
[the remark this answers is not in the document]
Comment (kleebaum) [reply plugin-src/gitVisualization/GitVisualizationToolbarView.ts:123]: renamed to shapeCommitToggle for a single commit :)

File: plugin-src/gitVisualization.ts (hunk 3 above)
Comment (kleebaum) [reply plugin-src/gitVisualization.ts:21]: application menu entries are now removed since they were redundant to the sidebar functionality

**Pull request merged**